libs/workspace/src/schematics/dt-component-example/index.ts

```diff
@@ -0,0 +1,258 @@
+
+    const importChange = addImport(
+      sourceFile,
+      options.exampleComponent.component,
+      options.exampleRoute,
+      modulePath,
+    );
+
+    // find last module in the array, and add new module
+    const modulesDeclaration = findNodes(
+      sourceFile,
+      ts.SyntaxKind.VariableDeclaration,
+    ).find(
+      (node: ts.VariableDeclaration) =>
+        node.name.getText() === options.exampleComponent.modulesConstant,
+    ) as ts.VariableDeclaration;
+    const modulesElements = (modulesDeclaration.initializer as ts.ArrayLiteralExpression)
+      .elements;
+    const lastElement = modulesElements[modulesElements.length - 1];
+    const end = modulesElements.hasTrailingComma
+      ? lastElement.getEnd() + 1
+      : lastElement.getEnd();
+    const routesChange = new InsertChange(
+      modulePath,
+      end,
+      `${modulesElements.hasTrailingComma ? ' ' : ', '}${
+        options.exampleComponent.component
+      }`,
+    );
+
+    return commitChanges(host, [importChange, routesChange], modulePath);
+  };
+}
+
+function updateIndex(options: DtExampleExtendedOptions): Rule {
+  return (host: Tree) => {
+    // add export to index.ts
+    const indexPath = join(
+      'libs',
+      'examples',
+      'src',
+      strings.dasherize(options.component),
+      'index.ts',
+    );
+    const sourceIndexFile = getSourceFile(host, indexPath);
+    const exportChange = addExport(
+      sourceIndexFile,
+      options.exampleRoute,
+      indexPath,
+    );
+
+    return commitChanges(host, exportChange, indexPath);
+  };
+}
+
+export default function (options: DtComponentExampleOptions): Rule {
+  return async (tree: Tree) => {
+    const exampleId = `${strings.dasherize(
+      options.component,
+    )}-${strings.dasherize(options.name)}`;
+    const extendedOptions: DtExampleExtendedOptions = {
+      ...options,
+      componentSelector: `dt-${strings.dasherize(options.component)}`,
+      selector: `dt-example-${strings.dasherize(
+        options.name,
+      )}-${strings.dasherize(options.component)}`,
+      componentModule: generateComponentOptions(options),
+      exampleComponent: generateExampleComponentOptions(options),
+      examplesModule: `DtExamples${strings.classify(options.component)}Module`,
+      exampleRoute: `${exampleId}-example/${exampleId}-example`,
+    };
+
+    const templateSource = apply(url('./files'), [
+      template({
+        ...strings,
+        ...extendedOptions,
+      }),
+    ]);
+
+    const indexPath = normalize(
+      `libs/examples/src/${strings.dasherize(options.component)}/index.ts`,
+    );
+
+    const isNewComponent = !tree.exists(indexPath);
+
+    if (isNewComponent) {
+      tree.create(indexPath, generateIndexFileContent(extendedOptions));
+
+      const modulePath = normalize(
+        `libs/examples/src/${strings.dasherize(
+          options.component,
+        )}/${strings.dasherize(options.component)}-examples.module.ts`,
```

Comment:
NIT: reuse the dasherized variable instead of manipulating the string two times again.